php/libraries/NDB_BVL_Instrument.class.inc

```diff
@@ -779,7 +779,11 @@ abstract class NDB_BVL_Instrument extends NDB_Page
      */
     function _saveValues(array $values): void
     {
-        if (strrpos($this->testName, "_proband") === false) {
+        // Surveys do not ask for the metadata informatio nand thus date_taken is
+        // always null so there is no use in going into the savecandidateage function
+        if (strrpos($this->testName, "_proband") === false
+            && $this->DataEntryType === 'normal'
```

Comment:
@driusan I'm a bit unsure if $this->DataEntryType is set at this stage. it seems to only be set in the `display()` method so.... I don't think it has any value here? Unless I'm missing something

Reply:
@ridz1208 i think this check is good here, the display method gets called on loading an instrument, so DataEntryType is set. it's not necessarily required though cz the first line of _saveCandidateAge checks if $values['Date_taken'] is empty

Reply:
fair point